### PR TITLE
chore: Adds  unused and misspell lint checks

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

This PR adds the `unused`, `typecheck`, and `misspell` lint checks and makes the necessary changes to satisfy them. We also removed the deprecated linters like `deadcode`.

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: .golangci.yml [hunk 1]
```diff
@@ -223,25 +223,21 @@ linters:
     # - govet
     # - errcheck
     # - staticcheck
-    # - unused
+    - unused
     # - gosimple
-    # - structcheck
-    # - varcheck
```
Comment (gab-arrobo) [inline .golangci.yml:229]: Why are these deleted? Are not we planning to enable them later?

File: .golangci.yml [hunk 2]
```diff
@@ -223,25 +223,21 @@ linters:
...
     - ineffassign
-    # - deadcode
```
Comment (gab-arrobo) [inline .golangci.yml:231]: Why is this deleted? Are not we planning to enable it later?

File: .golangci.yml [hunk 3]
```diff
@@ -223,25 +223,21 @@ linters:
...
-    # - typecheck
+    - typecheck
     # Additional
     # - lll
     - godox
     # - gomnd
     # - goconst
     # - gocognit
-    # - maligned
```
Comment (gab-arrobo) [inline .golangci.yml:239]: Why is this deleted? Are not we planning to enable it later?

File: .golangci.yml (hunk 2 above)
Comment (gruyaume) [reply .golangci.yml:231]: No, deadcode is a deprecated linter (as mentioned in the PR description) and was replaced with unused.

File: .golangci.yml (hunk 1 above)
Comment (gruyaume) [reply .golangci.yml:229]: Please see the answer to the `deadcode` removal.

File: .golangci.yml (hunk 3 above)
Comment (gruyaume) [reply .golangci.yml:239]: Maligned is also deprecated (replaced by govet)

**Review by gab-arrobo — APPROVED** [review]
+1 (tested changes locally)

**Pull request merged**